frappe/integrations/doctype/nextcloud_settings/nextcloud_settings.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+from frappe import _
+from frappe.utils.backups import new_backup
+from frappe.utils.background_jobs import enqueue
+from frappe.integrations.offsite_backup_utils import get_latest_backup_file, send_email, validate_file_size, generate_files_backup
+from frappe.utils import cint, get_backups_path
+
+import requests
+import os
+from rq.timeouts import JobTimeoutException
+from urllib.parse import urlparse
+
+class NextCloudSettings(Document):
+
+	def validate(self):
+		if not self.enabled:
+			return
+
+@frappe.whitelist()
+def take_backup():
+	"""Enqueue longjob for taking backup to nextcloud"""
+	enqueue("frappe.integrations.doctype.nextcloud_settings.nextcloud_settings.start_backup", queue='long', timeout=1500)
+	frappe.msgprint(_("Queued for backup. It may take a few minutes to an hour."))
+
+def take_backups_daily():
+	take_backups_if("Daily")
+
+def take_backups_weekly():
+	take_backups_if("Weekly")
+
+def take_backups_if(freq):
+	if frappe.db.get_value("NextCloud Settings", None, "backup_frequency") == freq:
+		start_backup()
+
+def start_backup():
+	backup = NextCloudController()
+	backup.take_backup_nextcloud()
+
+class NextCloudController():
+
+	def __init__(self, ignore_files=True, db_response="", site_config_response="",
+		did_not_upload=[], error_log=[], path_provided = False, nextcloud_settings=None, nextcloud_url=None,
+		webdav_url=None, email=None, password=None, base_url=None, url=None, session=None):
+		self.ignore_files = ignore_files
+		self.db_response = db_response
+		self.site_config_response = site_config_response
+		self.did_not_upload = did_not_upload
+		self.error_log = error_log
+		self.path_provided = path_provided
+		self.nextcloud_settings = nextcloud_settings
+		self.nextcloud_url = nextcloud_url
+		self.webdav_url = webdav_url
+		self.email = email
+		self.password = password
+		self.base_url = base_url
+		self.url = url
+		self.session = session
+
+	def take_backup_nextcloud(self, retry_count=0, upload_db_backup=True):
+		try:
+			if cint(frappe.db.get_value("NextCloud Settings", None, "enabled")):
```

Comment:
You already have the settings in the class as a property, why are you running queries again? The reason I suggested going object oriented was to reduce these and share data across functions